### ISSUE TITLE
Prometheus metric support for OpenAI endpoints

### ISSUE THREAD

**Issue by tylerhyang** [kserve/kserve#3983]
/kind feature

**Describe the solution you'd like**
Expose metrics for the OpenAI endpoints for models using the huggingfaceserver runtime.  The following metrics to be exposed should include:
TTFT (Time to First Token)
TPOT (Time Per Output Token)
Total Generated Tokens
Total Prompt Tokens
Request Latency

For both OIP and OpenAI endpoints, we should also expose the following metrics:
Total Successful Requests
Total Failed Requests

**Links to the design documents:**
RFC Template: https://docs.google.com/document/d/1IP1_fv7xg-9zERQmDRfcwVeDyBYEO6I_LU7nrAgdtu4/edit#heading=h.xw1gqgyqs5b
[Required, use the longer-form [design doc template] WIP after RFC feedback]

**Comment by yuzisun** [comment]
@tylerhyang these metrics are exposed already if you are using vllm backend, are you looking for the same set of metrics when using huggingface backend?

**Comment by tylerhyang** [comment]
> @tylerhyang these metrics are exposed already if you are using vllm backend, are you looking for the same set of metrics when using huggingface backend?

correct; this is particularly for the huggingface backend of the huggingfacerserver runtime
